src/Experiment.cpp

```diff
@@ -198,16 +199,24 @@ void Experiment::run_single()
 	}
 
 	auto start_time = std::chrono::high_resolution_clock::now();
-	pid_t workload;
-	if ((workload = fork())) {
+
+	if (settings::continuous_print_flag && settings::no_workload_flag) {
+		// do not fork if the continous print flag is set and we do not specify a workload.
 		sampler.start(std::max(-settings::before, std::chrono::milliseconds(0)));
-		waitpid(workload, NULL, 0);
+		std::promise<void>().get_future().wait(); // waits forever
```

Comment:
This is not optimal in my opinion, but I think waiting forever here is needed due to the program design. 

See: https://stackoverflow.com/a/42644441